File: model/Question.js
const mongoose = require('mongoose');

const questionSchema = new mongoose.Schema({
    userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
    content: { type: String, required: true },
    image: { type: String }, 
    status: {
        type: String,
        enum: ['pending', 'approved', 'rejected'],
        default: 'pending'
    },
    likes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }], 
    dislikes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }], // Thêm trường dislike
    reports: [{ // Thêm trường báo cáo
        userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
        reason: String,
        createdAt: { type: Date, default: Date.now }
    }],
    comments: [{
        userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
        text: String,
        likes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }], // Like cho comment
        dislikes: [{ type: mongoose.Schema.Types.ObjectId, ref: 'User' }], // Dislike cho comment
        reports: [{ 
            userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
            reason: String,
            createdAt: { type: Date, default: Date.now }
        }],
        createdAt: { type: Date, default: Date.now }
    }]
}, { timestamps: true });

module.exports = mongoose.model('Question', questionSchema);

File: app/src/main/java/com/example/doan_mau/model/Question.java
package com.example.doan_mau.model;

import java.util.List;

public class Question {
    private String id;
    private String userName;
    private String content;
    private int likes;
    private int dislikes;
    private List<Comment> comments;

    public Question(String id, String userName, String content, int likes, int dislikes) {
        this.id = id;
        this.userName = userName;
        this.content = content;
        this.likes = likes;
        this.dislikes = dislikes;
    }

    public String getId() { return id; }
    public String getUserName() { return userName; }
    public String getContent() { return content; }
    public int getLikes() { return likes; }
    public int getDislikes() { return dislikes; }
}